packages/experimental/Badge/src/CounterBadge/useCounterBadge.ts

```diff
@@ -0,0 +1,21 @@
+import { CounterBadgeProps, CounterBadgeInfo } from './CounterBadge.types';
+
+export const useCounterBadge = (props: CounterBadgeProps): CounterBadgeInfo => {
+  const DEFAULT_OVERFLOW_COUNT = 99;
+  const { dot, overflowCount = DEFAULT_OVERFLOW_COUNT, showZero, count, ...rest } = props;
+
+  const _count = count && count > overflowCount ? `${overflowCount}+` : `${count}`;
+  const showBadge = count !== 0 || (count === 0 && !!showZero);
+
+  return {
+    props: {
+      dot: !!dot,
```

Comment:
nit: It doesn't look like this is necessary. Imo you don't always need to !! a boolean to ensure that undefined becomes false since most operators handle it ok.